pkg/bootkube/bootkube.go

```diff
@@ -129,13 +130,27 @@ func (b *bootkube) Run() error {
 			errch <- err
 		}
 	}()
+	var etcdServiceIP string
+	var err error
 	if b.selfHostedEtcd {
-		requiredPods = append(requiredPods, "etcd-operator")
+		etcdServiceIP, err = detectEtcdIP(b.assetDir)
+		if err != nil {
+			return err
+		}
 	}
-	go func() { errch <- WaitUntilPodsRunning(requiredPods, assetTimeout, b.selfHostedEtcd) }()
-
+	go func() {
+		if b.selfHostedEtcd {
+			requiredPods = append(requiredPods, "etcd-operator")
+			if err := WaitUntilPodsRunning(requiredPods, assetTimeout); err != nil {
+				errch <- err
```

Comment:
my mistake in example, should also return here after pushing err onto chan (not a big deal, as the program will exit anyway - but may as well not try and execute the migration).